src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc

```diff
@@ -353,6 +353,13 @@ void CoilCoolingDXCurveFitOperatingMode::CalcOperatingMode(EnergyPlus::EnergyPlu
     }
     outletNode.Temp = Psychrometrics::PsyTdbFnHW(outletNode.Enthalpy, outletNode.HumRat);
 
+    // Check for saturation error and modify temperature at constant enthalpy
+    Real64 tsat = Psychrometrics::PsyTsatFnHPb(state, outletNode.Enthalpy, inletNode.Press, RoutineName);
+    if (outletNode.Temp < tsat) {
+        outletNode.Temp = tsat;
+        outletNode.HumRat = Psychrometrics::PsyWFnTdbH(state, tsat, outletNode.Enthalpy);
+    }
+
```

Comment:
This may be the wrong spot to check Tsat. Above at line 330 bypass mixing is calculated, then at line 350 cycling fan Tout is calculated, both of these pull back/right into the psyc chart but will be using Tout<Tsat for those calculations. So move this above 330?

Reply:
There's already a saturation check at [line 338](https://github.com/NREL/EnergyPlus/blob/4c061e4ee1ead44d8a6bd8b9e4a808beb1e8a467/src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc#L337-L341) so I think that's covered.

Plus there's another existing check at [line 382](https://github.com/NREL/EnergyPlus/blob/4c061e4ee1ead44d8a6bd8b9e4a808beb1e8a467/src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc#L382-L386).

But these are all within a single operating mode. Perhaps there's need for yet another check at the performance level when running with more than one operating mode?

Reply:
Those checks are likely never acted on unless bypass fraction is very small since when mixing bypassed air the outlet condition moves toward the inlet condition (same for cycling fan) and Tout will likely never be < Tsat. It's the mixing calculation prior to line 338 and 382 that I am concerned about. Seems that this code change could be moved to the bottom of CalcSpeedOutput? so we know the outlet condition is valid on subsequent calculations.

Reply:
Another thought, just to muddy the waters, and a total side bar. This check, wherever it is, is a step function change in delivered capacity (sensible or latent, not total) and could impact RootSolver. I saw the same RootSolver issue with latent degradation where SHR=1 from PLR=0 to about PLR=0.5. RootSolver got confused on how to converge and sometimes hit max iteration. RootSolver iterates well on a smooth function.

Reply:
looks like line 387 below here:

    outletNode.Temp = correctedTemp;

can be deleted, it gets overwritten just after there.

Reply:
I think lines 365-372 should be moved into the if (fanOpMode == DataHVACGlobals::ContFanCycCoil) block. Hasn't Tsat already be tested before this point? and is only needed if fanOpMode = ContFanCycCoil? Why do this every time?

Reply:
Is the reason to do this all the time because the outlet temp could be over the sat curve coming out of CalcSpeedOutput? If so, why not put it there?

Reply:
Here is why I think a Tsat check should be in CalcSpeedOutput. At line 341 the outlet condition is adjusted but is based on outletNode.HumRat that has not yet been corrected. And at line 361 the outlet condition is adjusted but is also based on outletNode.HumRat. Should we not be using a valid outlet condition in these 2 places? So Tsat needs to be tested before this point, and the best choice would be CalcSpeedOutput. The Tsat test is also needed here inside the contfan if block.

Reply:
I had convinced myself earlier that `CalcSpeedOutput` didn't need this check because of the way it arrives at the leaving conditions, but I'll add it anyway just to see what happens. Ultimately this branch was started in an attempt to align with the old DX coil(s), because I had seen one place where the old multi-speed coil had this correction and the new coil did not. Now this branch may have gone beyond or may have superfluous (and somewhat costly) checks. It seems all of this double-checking shouldn't be necessary, but we've got multiple layers of partial results being mixed together (face area, speeds, and cycling coil with continuous fan, and latent degradation). Do we check at each step or just wait until the end and do a single check? Going back to the old DX coil to see how many places this check is present.

Reply:
And the other check has been moved into the `ContFanCyclCoil` if block.

Reply:
I am thinking in CalcSpeedOutput, depending on how close the inlet condition is to the sat curve, that when the coil ADP is found, which is on the sat curve, that the actual outlet condition based on the bypass factor could be slightly to the left of the sat curve (in the 0.1C range).